platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/platformstatus/PlatformStatusStateMachine.java

```diff
@@ -0,0 +1,115 @@
+package com.swirlds.common.system.platformstatus;
+
+import static com.swirlds.logging.LogMarker.EXCEPTION;
+import static com.swirlds.logging.LogMarker.PLATFORM_STATUS;
+
+import com.swirlds.common.notification.NotificationEngine;
+import com.swirlds.common.notification.listeners.PlatformStatusChangeListener;
+import com.swirlds.common.notification.listeners.PlatformStatusChangeNotification;
+import com.swirlds.common.system.platformstatus.statuslogic.PlatformStatusLogic;
+import com.swirlds.common.system.platformstatus.statuslogic.StatusLogicFactory;
+import com.swirlds.common.time.Time;
+import com.swirlds.logging.payloads.PlatformStatusPayload;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Duration;
+import java.util.Objects;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * The platform status state machine
+ */
+public class PlatformStatusStateMachine {
+    private static final Logger logger = LogManager.getLogger(PlatformStatusStateMachine.class);
+
+    /**
+     * A source of time
+     */
+    private final Time time;
+
+    /**
+     * The platform status config
+     */
+    private final PlatformStatusConfig config;
+
+    /**
+     * For passing notifications between the platform and the application.
+     */
+    private final NotificationEngine notificationEngine;
+
+    /**
+     * The object containing the state machine logic for the current status
+     */
+    private PlatformStatusLogic currentStatusLogic;
+
+    /**
+     * Constructor
+     *
+     * @param time               a source of time
+     * @param config             the platform status config
+     * @param notificationEngine the notification engine
+     */
+    public PlatformStatusStateMachine(
+            @NonNull final Time time,
+            @NonNull PlatformStatusConfig config,
+            @NonNull final NotificationEngine notificationEngine) {
+
+        this.time = Objects.requireNonNull(time);
+        this.config = Objects.requireNonNull(config);
+        this.notificationEngine = Objects.requireNonNull(notificationEngine);
+
+        this.currentStatusLogic = StatusLogicFactory.createStatusLogic(PlatformStatus.STARTING_UP, time, config);
+    }
+
+    /**
+     * Process a platform status action.
+     * <p>
+     * Repeated calls of this method cause the platform state machine to be traversed
+     *
+     * @param action the action to process
+     */
+    public void processStatusAction(@NonNull final PlatformStatusAction action) {
+        Objects.requireNonNull(action);
+
+        final PlatformStatus newStatus;
+        try {
+            newStatus = currentStatusLogic.processStatusAction(action);
+        } catch (final IllegalArgumentException e) {
+            logger.error(EXCEPTION.getMarker(), e.getMessage());
+            return;
+        }
+
+        // if status didn't change, there isn't anything to do
+        if (newStatus.equals(currentStatusLogic.getStatus())) {
```

Comment:
Use `==` to compare enum values. Enums are not regular java objects, `.equals()` is not necessary.